File: mdetect/src/main/java/me/myatminsoe/mdetect/MMToast.java
package me.myatminsoe.mdetect;

import android.content.Context;
import android.widget.Toast;

public class MMToast {

    public static final int LENGTH_SHORT = Toast.LENGTH_SHORT;
    public static final int LENGTH_LONG = Toast.LENGTH_LONG;

    public static Toast makeText(Context context, CharSequence message, int duration) {
        if (!MDetect.isUnicode()) message = Rabbit.uni2zg(message.toString());
        return Toast.makeText(context, message, duration);
    }

    public static Toast makeText(Context context, int resId, int duration) {
        String message = context.getResources().getString(resId);
        if (!MDetect.isUnicode()) message = Rabbit.uni2zg(message);
        return Toast.makeText(context, message, duration);
    }
}

File: mdetect/src/main/java/me/myatminsoe/mdetect/MMToast.kt
package me.myatminsoe.mdetect

import android.content.Context
import android.widget.Toast

object MMToast {

    private var toast: Toast? = null

    fun showShortToast(context: Context, message: CharSequence) {
        toast?.cancel()
        toast = Toast.makeText(context, MDetect.getText(message.toString()), Toast.LENGTH_SHORT)
        toast!!.show()
    }

    fun showLongToast(context: Context, message: CharSequence) {
        toast?.cancel()
        toast = Toast.makeText(context, MDetect.getText(message.toString()), Toast.LENGTH_LONG)
        toast!!.show()
    }
}
